src/adb_tcp_connexion.rs

```diff
@@ -201,10 +201,12 @@ impl AdbCommandProvider for AdbTcpConnexion {
             match tcp_stream.read(&mut buffer) {
                 Ok(size) => {
                     if size == 0 {
-                        return Ok(());
+                        return Ok("".to_string());
                     } else {
-                        print!("{}", String::from_utf8(buffer.to_vec())?);
+                        let output = String::from_utf8(buffer.to_vec())?;
+                        print!("{}", &output);
                         std::io::stdout().flush()?;
+                        return Ok(output);
```

Comment:
This only returns the output received in the the first `tcp_stream.read()` call. This could be as little as a single byte if unlucky.

In situations where you need to process the output, and not just print it then I imagine it could be good to have an alternative API, something like `shell_command_stream()` that can return a Stream instead, maybe just return the TcpStream.